ath79: add support for DomyWifi DW33D

shanpo opened the pull request:
This commit porting the device from ar71xx to the ath79 target
and modifies the partition layout.
1. Firmware is installed to nand flash.
2. Modify the uboot-env parameter to boot from the nand flash.
3. The kernel size is extended to 5M.
4.nor flash retains the oem firmware.

oem partition layout
dev:    size   erasesize  name
mtd0: 00040000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
mtd2: 00e30000 00010000 "rootfs"
mtd3: 00170000 00010000 "kernel"
mtd4: 00010000 00010000 "art"
mtd5: 00f90000 00010000 "firmware"
mtd6: 06000000 00020000 "rootfs_data"
mtd7: 02000000 00020000 "backup"

new partition layout
dev:    size   erasesize  name
mtd0: 00040000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
mtd2: 00fa0000 00010000 "oem-firmware"
mtd3: 00010000 00010000 "art"
mtd4: 00500000 00020000 "kernel"
mtd5: 07b00000 00020000 "ubi"

oem u-boot-env
bootcmd=bootm 0x9fe80000

new u-boot-env
bootcmd=nboot 0x8050000 0;bootm || bootm 0x9fe80000

installation guide
1.Upload openwrt firmware to the device
scp openwrt-snapshot-r11365-df60a0852c-ath79-nand-domywifi_dw33d-\
squashfs-factory.bin root@192.168.10.1:/tmp
2.modify uboot-env.
ssh login to the device:
fw_setenv bootcmd 'nboot 0x8050000 0;bootm || bootm 0x9fe80000'
Run the fw_printenv command to check if the settings are correct.
3.Write openwrt firmware.
ssh login to the device:
mtd -r write /tmp/openwrt-snapshot-r11365-df60a0852c-ath79-nand-\
domywifi_dw33d-squashfs-factory.bin /dev/mtd6
The device will restart automatically and the openwrt firmware
installation is complete.

Restore oem firmware.just erase the kernel partition and the ubi partition.
ssh login to the device:
mtd erase /dev/mtd4
mtd -r erase /dev/mtd5
Reboot the device

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,215 @@
+			label = "ubi";
+			reg = <0x500000 0x7b00000>;
+		};
+
```
Remove empty line.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,215 @@
+&wmac {
+	status = "okay";
+	mtd-cal-data = <&art 0x1000>;
+	mtd-mac-address = <&art 0x12>;
```
0xc?

adrianschmutzler replied:
Looked it up in ar71xx, it's indeed "12" in decimal notation, so it should be 0xc in hex notation.
Please also check whether there is a valid MAC address in art 0x1002.

shanpo replied:
I set an error value, which should be 0xc

adrianschmutzler commented on target/linux/ath79/image/nand.mk:
```diff
@@ -43,6 +43,22 @@ define Device/aerohive_hiveap-121
+  ATH_SOC := qca9558
+  DEVICE_VENDOR := DomyWifi
+  DEVICE_MODEL := DW33D
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-storage kmod-usb-ledtrig-usbport kmod-ath10k-ct ath10k-firmware-qca988x-ct
```
Remove kmod-usb-core, it is pulled by the other packages.

adrianschmutzler commented on target/linux/ath79/nand/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:
```diff
@@ -3,13 +3,16 @@
 	case $board in
+	domywifi,dw33d)
+		caldata_extract "art" 0x5000 0x844
+		ath10kcal_patch_mac $(mtd_get_mac_binary art 0x12)
```
-> ath10k_patch_mac

adrianschmutzler replied:
Ah, so this could really be 0x12 in this case. Please check whether there is a valid MAC address in art 0x5006.
I would be interested to see a full hexdump of the art partition. Refer to https://openwrt.org/docs/guide-developer/mac.address for further instructions (if necessary).

shanpo replied:
‘mtd_get_mac_binary art 0x12’Can return the correct MAC address,

adrianschmutzler replied:
Okay, still please provide us with an overview of the MAC addresses assignment on stock firmware and the art location. This could be added to the commit message later. What about 0x1002 and 0x5006?

shanpo replied:
eth0 gets the mac from the art address 0x0, eth1 is 0x6, 0xc is the mac address of the ath9k device, and 0x12 is the ath10k device. These mac addresses are identical to the oem firmware.
0x1002 and 0x5006 are not valid mac addresses.
Please forgive me for not knowing English.

adrianschmutzler replied:
Okay, perfect. Please add this information to the commit message, e.g.
MAC address overview:
eth0: 0x0
eth1: 0x6
ath9k: 0xc
ath10k: 0x12
No valid addresses in 0x1002 and 0x5006. All addresses match the OEM firmware.
```

adrianschmutzler replied:
Don't forget to change the function name.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,215 @@
+		led-upgrade = &status;
+	};
+
+	gpio_leds: leds {
```
As you don't use it for anything, remove the label.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,215 @@
+		};
+	};
+
+	gpio_keys: keys {
```
As you don't use it for anything, remove the label.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,215 @@
+		internet {
+			label = "dw33d:blue:internet";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			default-state = "on";
```
Why default=on on internet?

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,215 @@
+			label = "dw33d:blue:usb0";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&hub_port0>;
+		};
```
So, why did you change from "mmc" to "usb0"?
(This will need migration for labels, too.)

shanpo replied:
This device does not have an mmc flasher, but a usb card reader. I hope to avoid ambiguity.

981213 replied:
If the usb card reader is already on board and this led is for mmc activity I think it should be named mmc instead.

shanpo replied:
Ok, continue to use the mmc name.

adrianschmutzler replied:
But then it shouldn't trigger for the usb either?

shanpo replied:
The card reader is embedded in the circuit board and occupies usb0.Panel led display properly.

adrianschmutzler commented on target/linux/ath79/nand/base-files/etc/board.d/02_network:
```diff
@@ -11,6 +11,10 @@ ath79_setup_interfaces()
 		;;
+	domywifi,dw33d)
+		ucidef_add_switch "switch0" \
+			"0@eth1" "2:lan" "3:lan" "4:lan" "5:lan" "6@eth0" "1:wan"
```
Is this affected by eth0/eth1 swap or why are you deviating from ar71xx here?

shanpo replied:
Yes, the eth0/ethe port changes.

adschm reviewed (commented):
That should be enough for today ;-)

What about
```
	ath79_nfc_set_ecc_mode(AR934X_NFC_ECC_HW);
	ath79_register_nfc();
```
in ar71xx mach files?

981213 commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,215 @@
+	status = "okay";
+};
+
+&gpio {
```
gpio is already enabled in qca9557.dtsi. this node can be dropped.

981213 commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,215 @@
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
```
If both eth nodes uses fixed-link there's no reference to the phy0 node defined above and the switch isn't configurable. Remove this fixed-link node and add the following line here:
```
phy-handle = <&phy0>;
```

981213 commented on target/linux/ath79/nand/base-files/etc/board.d/01_leds:
```diff
@@ -8,6 +8,13 @@ board=$(board_name)
 case "$board" in
+domywifi,dw33d)
+	ucidef_set_led_usbport "usb0" "USB0" "$boardname:blue:usb0" "usb1-port1"
+	ucidef_set_led_usbport "usb1" "USB1" "$boardname:blue:usb1" "usb2-port1"
```
usb triggers could be set in dts instead.

981213 reviewed (changes requested):
Hi! Some comments below:

shanpo commented:
I have reorganized the code and tested

981213 commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,209 @@
+
+		partition@500000 {
+			label = "ubi";
+			reg = <0x500000 0x7b00000>;
```
There's a backup partition in ar71xx. Why is that squashed into this ubi partition here? If vendor stores anything in that partition it should be left untouched. Generally we should not use spaces other than  the vendor firmware and data partition.

shanpo replied:
The backup partition is the space that the vendor firmware provides to the user, and the vendor system does not depend on this partition. Some users recommend merging them into rootfs to accommodate applications. Of course, if this is not compliant, keep this partition.

981213 replied:
You shouldn't touch partitions that vendor reserves. Changing u-noot-env is okay because it can be easily reverted. This erases an entire partition and it can't be reversed without a partition backup.

shanpo replied:
Ok, I am restoring the backup partition.

981213 commented on target/linux/ath79/image/nand.mk:
```diff
@@ -43,6 +43,22 @@ define Device/aerohive_hiveap-121
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  UBINIZE_OPTS := -E 5
+  SUPPORTED_DEVICES += dw33d
```
Your version is not compatible with dw33d support in ar71xx at all. This SUPPORTED_DEVICES line should be dropped.

adrianschmutzler commented on target/linux/ath79/image/nand.mk:
```diff
@@ -45,6 +45,21 @@ define Device/aerohive_hiveap-121
+  IMAGES += factory.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi
+endef
```
IMAGE_SIZE is missing here.

adrianschmutzler commented on target/linux/ath79/nand/base-files/lib/upgrade/platform.sh:
```diff
@@ -13,6 +13,12 @@ platform_do_upgrade() {
+	netgear,wndr4300|\
+	zyxel,nbg6716)
+		nand_do_upgrade "$1"
+		;;
```
I've made nand_do_upgrade default here, so you can drop changes to this file entirely.

adrianschmutzler commented:
Please squash your changes into a single commit (git rebase -i ...). Your first commit lacks the signed-off-by at the moment.

adrianschmutzler commented:
Signed-off-by should be at the end of your commit message.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,209 @@
+	leds {
+		compatible = "gpio-leds";
+
+		system: system {
```
Please use "led_system" for the label:
`led_system: system {`
and update aliases accordingly.

shanpo committed:
ath79: add support for DomyWifi DW33D

This commit porting the device from ar71xx to the ath79 target
and modifies the partition layout.
1. Firmware is installed to nand flash.
2. Modify the uboot-env parameter to boot from the nand flash.
3. The kernel size is extended to 5M.
4.nor flash retains the oem firmware.

oem partition layout
dev:    size   erasesize  name
mtd0: 00040000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
mtd2: 00e30000 00010000 "rootfs"
mtd3: 00170000 00010000 "kernel"
mtd4: 00010000 00010000 "art"
mtd5: 00f90000 00010000 "firmware"
mtd6: 06000000 00020000 "rootfs_data"
mtd7: 02000000 00020000 "backup"

new partition layout
dev:    size   erasesize  name
mtd0: 00040000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
mtd2: 00fa0000 00010000 "oem-firmware"
mtd3: 00010000 00010000 "art"
mtd4: 00500000 00020000 "kernel"
mtd5: 05b00000 00020000 "ubi"
mtd6: 02000000 00020000 "oem-backup"

oem u-boot-env
bootcmd=bootm 0x9fe80000

new u-boot-env
bootcmd=tftp 0x84000000 dw33d.bin;bootm 0x84000000 || nboot \
0x8050000 0
This parameter first loads the ram firmware (dw33d.bin) from \
the TFTP server. If it fails, it loads the firmware from Nand.

MAC address overview:
All mac addresses are stored in the art partition.
eth0: 0x0
eth1: 0x6
ath9k: 0xc
ath10k: 0x12
No valid addresses in 0x1002 and 0x5006. All addresses match the OEM firmware.

Install from oem firmware.
Enable ssh service:
Connect to the router web, click professional, click system-startup,\
 and add dropbear in the local startup input box. Click system-admins\
tration, delete ssh-key, and replace your ssh pub key. Restart the router.
1.Upload openwrt firmware to the device
scp openwrt-snapshot-r11365-df60a0852c-ath79-nand-domywifi_dw33d-\
squashfs-factory.bin root@192.168.10.1:/tmp
2.modify uboot-env.
ssh login to the device:
fw_setenv bootcmd 'nboot 84000000 0;bootm || bootcmd=bootm 0x9fe80000'
Run the fw_printenv command to check if the settings are correct.
3.Write openwrt firmware.
ssh login to the device:
mtd -r write /tmp/openwrt-snapshot-r11365-df60a0852c-ath79-nand-\
domywifi_dw33d-squashfs-factory.bin /dev/mtd6
The device will restart automatically and the openwrt firmware
installation is complete.

Restore oem firmware.just erase the kernel partition and the ubi partition.
ssh login to the device:
mtd erase /dev/mtd4
mtd -r erase /dev/mtd5
Reboot the device

Signed-off-by: WeiDong Jia <jwdsccd@gmail.com>

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,214 @@
+
+&wmac {
+	status = "okay";
+	mtd-cal-data = <&art 0x1000>;
```
Please add an empty line after status.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,214 @@
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
```
Please add an empty line after status.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,214 @@
+		};
+
+		usb1 {
+			label = "dw33d:blue:usb1";
```
Since the previous led was renamed to mmc, you should rename this one "usb1" -> "usb".
This will also match the ar71xx led label then.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,214 @@
+			linux,default-trigger = "usbport";
+		};
+
+		wlan-2g {
```
I would personally use "wlan2g", but that's a matter of taste. The label can't be changed, as it's determined by ar71xx.

adrianschmutzler commented on target/linux/ath79/image/nand.mk:
```diff
@@ -45,6 +45,22 @@ define Device/aerohive_hiveap-121
+  DEVICE_MODEL := DW33D
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-storage kmod-usb-ledtrig-usbport kmod-ath10k-ct ath10k-firmware-qca988x-ct
+  KERNEL_SIZE := 5120k
+  IMAGE_SIZE := 96m
```
Please use 98304k here, to be consistent with other values.

adrianschmutzler commented on target/linux/ath79/nand/base-files/etc/board.d/01_leds:
```diff
@@ -8,6 +8,13 @@ board=$(board_name)
+	ucidef_set_led_usbport "usb1" "USB1" "$boardname:blue:usb1" "usb2-port1"
+	ucidef_set_led_netdev "internet" "INTERNET" "$boardname:blue:internet" "eth0"
+	ucidef_set_led_wlan "wlan2g" "WLAN-2.4G" "$boardname:blue:wlan-2g" "phy1tpt"
+	ucidef_set_led_wlan "wlan5g" "WLAN-5G" "ath10k-phy0" "phy0tpt"
```
What happened to the internet LED?

adrianschmutzler replied:
Despite, I wonder whether it would be possible to set up ath10k-leds similar to what is done with ath9k-leds for wdr4900-v2:
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9558_tplink_tl-wdr4900-v2.dts#L60
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9558_tplink_tl-wdr4900-v2.dts#L101

But don't hit me if it won't work.

shanpo replied:
> Despite, I wonder whether it would be possible to set up ath10k-leds similar to what is done with ath9k-leds for wdr4900-v2:
> https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9558_tplink_tl-wdr4900-v2.dts#L60
> https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9558_tplink_tl-wdr4900-v2.dts#L101
> 
> But don't hit me if it won't work.

Hello, it is completely feasible to define wlan5g led in dts.

adrianschmutzler commented on target/linux/ath79/nand/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:
```diff
@@ -9,6 +9,10 @@ board=$(board_name)
 	case $board in
+	domywifi,dw33d)
+		caldata_extract "art" 0x5000 0x844
+		ath10k_patch_mac $(mtd_get_mac_binary art 0x12)
```
With mtd_get_mac_binary being used here, can this be replaced with mtd_mac_address in DTS? Or is this not possible with ath10k?

shanpo replied:
I tried mtd_mac_address but could not get the correct mac address.

adrianschmutzler replied:
Note that is has to be a property of wifi@0,0:
```
&pcie0 {
   wifi@0,0 {
      ...
      mtd-mac-address = <&art 0x12>;
   };
};
```
(No offense if you already did it correctly...)

shanpo replied:
I don't know how to use mtd_mac_address to get the address in dts, I will try again, thank you.

adrianschmutzler replied:
Note that it might still not work, but I'm really interested to find out.

shanpo replied:
`&pcie0 {
	status = "okay";

	wifi@0,0 {
		compatible = "qcom,ath10k";
		reg = <0 0 0 0 0>;
		mtd-mac-address = <&art 0x12>;
	};
};`

Still can't get the correct mac address

adrianschmutzler replied:
Okay, thanks for testing!

shanpo commented:
I fixed the swatch configuration error, now eth0/eth1 is the same as the ar71xx target configuration.

adrianschmutzler commented:
> I fixed the swatch configuration error, now eth0/eth1 is the same as the ar71xx target configuration.

Please provide some more details on that (how was it before, how is it now, and why is the current solution correct). There are several devices where eth0/eth1 are indeed swapped now (when configured correctly) compared to ar71xx.

shanpo commented:
> > I fixed the swatch configuration error, now eth0/eth1 is the same as the ar71xx target configuration.
> 
> Please provide some more details on that (how was it before, how is it now, and why is the current solution correct). There are several devices where eth0/eth1 are indeed swapped now (when configured correctly) compared to ar71xx.

Mdio node qca,ar8327-initvals configuration parameters:
`	0x04 0x87600000 /* PORT0 PAD MODE CTRL */
	0x0c 0x00080080 /* PORT6 PAD MODE CTRL */`
Switching the 0x04/0x0c parameter can change the mapping between the switch PORT and eth, as long as the switch vlan is configured as:
VLAN 1:
        vid: 1
        ports: 2 3 4 5 6t 
VLAN 2:
        vid: 2
        ports: 0t 1
The device works fine, but the lan/wan physical device name will change in both configurations, so consider the same as the original ar71xx target.

shanpo commented:
Update wlan5g led, now defined in dts.

981213 commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,229 @@
+		compatible = "qcom,ath10k";
+		reg = <0 0 0 0 0>;
+		#gpio-cells = <2>;
+		gpio-controller;
```
It doesn't work this way. ath10k doesn't support gpio-controller at all.

shanpo replied:
Yes, I deleted the relevant code, 5g led is still controlled.

981213 replied:
Correct. ath10k leds will use its tpt trigger by default as you can see in this patch: https://github.com/openwrt/openwrt/blob/master/package/kernel/mac80211/patches/ath/975-ath10k-use-tpt-trigger-by-default.patch

adrianschmutzler commented on target/linux/ath79/image/nand.mk:
```diff
@@ -45,6 +45,22 @@ define Device/aerohive_hiveap-121
 TARGET_DEVICES += aerohive_hiveap-121
 
+define Device/domywifi_dw33d
+  ATH_SOC := qca9558
```
Please rename this to "SOC" due to 95030ac0b673165772daf27128359eaaf536d457.

Hexadecimaaal commented:
Is this ready to be merged into 19.07 and prepared to release in next minor update? It seems pretty complete to me. @981213 may the author of this close the requested change case now?

981213 commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,217 @@
+
+&nand {
+	status = "okay";
+	
```
trailing whitespace here

981213 commented on target/linux/ath79/dts/qca9558_domywifi_dw33d.dts:
```diff
@@ -0,0 +1,217 @@
+			label = "ubi";
+			reg = <0x500000 0x5b00000>;
+		};
+		
```
another trailing whitespace here

981213 commented on target/linux/ath79/nand/base-files/etc/board.d/02_network:
```diff
@@ -11,6 +11,11 @@ ath79_setup_interfaces()
 		ucidef_set_interface_lan "eth0"
 		;;
+	domywifi,dw33d)
+		ucidef_set_interfaces_lan_wan "eth1.1" "eth0.2"
```
This is unnecessary. ucidef_add_switch should handle lan and wan setup properly.

981213 reviewed (commented):
> new u-boot-env
> bootcmd=tftp 0x84000000 dw33d.bin;bootm 0x84000000 || nboot \
> 0x8050000 0
> This parameter first loads the ram firmware (dw33d.bin) from \
> the TFTP server. If it fails, it loads the firmware from Nand.

This is not a good choice. It opens a security hole where anyone could upload a firmware in lan when router is rebooted. And when there isn't a TFTP server available, this introduced a long boot delay when u-boot waits for a TFTP response. That's about half a minute if I remember correctly.
I've fixed this one in commit message and three small comments below.
It's merged into my staging tree at: https://git.openwrt.org/?p=openwrt/staging/981213.git
Could someone test it to see if it still works properly?

981213 commented:
BTW you need to sign-off with your realname instead of a nick name. Could you provide a proper sign-off line for me to fix the commit message?

adrianschmutzler commented on target/linux/ath79/image/nand.mk:
```diff
@@ -46,6 +46,22 @@ define Device/aerohive_hiveap-121
+  SOC := qca9558
+  DEVICE_VENDOR := DomyWifi
+  DEVICE_MODEL := DW33D
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-storage kmod-usb-ledtrig-usbport kmod-ath10k-ct ath10k-firmware-qca988x-ct
```
One could introduce a line break here ...

981213 replied:
Oh. I overlooked this. Fixed in my staging tree. Thanks for spotting it.

BTW It's weird that checkpatch.pl doesn't report this problem.

shanpo replied:
I reorganized the code and submission information, and modified the signature. But the main line upstream code seems to have a problem, do not compile ar934x_nand drivers, in addition, the firmware is operating normally.

adrianschmutzler commented on target/linux/ath79/image/nand.mk:
```diff
@@ -46,6 +46,22 @@ define Device/aerohive_hiveap-121
+  UBINIZE_OPTS := -E 5
+  IMAGES += factory.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-ubi | check-size $$$$(IMAGE_SIZE)
```
... and here.

shanpo commented:
kernel log:

```
[    0.000000] Linux version 4.19.106 (jwd@d6139) (gcc version 8.3.0 (OpenWrt GCC 8.3.0 r11734+136-b085e0586f)) #0 Tue Mar 3 21:37:28 2020
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019750 (MIPS 74Kc)
[    0.000000] MIPS: machine is DomyWifi DW33D
[    0.000000] SoC: Qualcomm Atheros QCA9558 ver 1 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 10000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000000fffffff]
[    0.000000] On node 0 totalpages: 65536
[    0.000000]   Normal zone: 576 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved ProTip! Add .patch or .diff to the end of URLs for Git’s plaintext views.
[    0.000000]   Normal zone: 65536 pages, LIFO batch:15
[    0.000000] random: get_random_bytes called from start_kernel+0x98/0x51c with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 64960
[    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 32768 (order: 5, 131072 bytes)
[    0.000000] Inode-cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 251344K/262144K available (4877K kernel code, 262K rwdata, 1160K rodata, 1308K init, 215K bss, 10800K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 720.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 5309056796 ns
[    0.000008] sched_clock: 32 bits at 360MHz, resolution 2ns, wraps every 5965232126ns
[    0.008296] Calibrating delay loop... 358.80 BogoMIPS (lpj=1794048)
[    0.074984] pid_max: default: 32768 minimum: 301
[    0.080091] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.087127] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.098895] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.109395] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.115997] pinctrl core: initialized pinctrl subsystem
[    0.121982] NET: Registered protocol family 16
[    0.133897] PCI host bridge /ahb/pcie-controller@180c0000 ranges:
[    0.140388]  MEM 0x0000000010000000..0x0000000011ffffff
[    0.145982]   IO 0x0000000000000000..0x0000000000000000
[    0.168504] PCI host bridge to bus 0000:00
[    0.172871] pci_bus 0000:00: root bus resource [mem 0x10000000-0x11ffffff]
[    0.180232] pci_bus 0000:00: root bus resource [io  0x0000]
[    0.186154] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.193380] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.201838] pci 0000:00:00.0: [168c:003c] type 00 class 0x028000
[    0.201892] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
[    0.201947] pci 0000:00:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
[    0.202024] pci 0000:00:00.0: supports D1
[    0.202032] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
[    0.203070] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    0.203094] pci 0000:00:00.0: BAR 0: assigned [mem 0x10000000-0x101fffff 64bit]
[    0.210920] pci 0000:00:00.0: BAR 6: assigned [mem 0x10200000-0x1020ffff pref]
[    0.219159] clocksource: Switched to clocksource MIPS
[    0.261824] NET: Registered protocol family 2
[    0.267147] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes)
[    0.275404] TCP established hash table entries: 2048 (order: 1, 8192 bytes)
[    0.282861] TCP bind hash table entries: 2048 (order: 1, 8192 bytes)
[    0.289653] TCP: Hash tables configured (established 2048 bind 2048)
[    0.296526] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.302790] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.309817] NET: Registered protocol family 1
[    0.314494] PCI: CLS 0 bytes, default 32
[    0.318825] Crashlog allocated RAM at address 0x3f00000
[    0.325670] workingset: timestamp_bits=14 max_order=16 bucket_order=2
[    0.338342] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.344593] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.367646] io scheduler noop registered
[    0.371867] io scheduler deadline registered (default)
[    0.377958] ar7200-usb-phy 18030000.usb-phy0: phy reset is missing
[    0.384639] ar7200-usb-phy 18030010.usb-phy1: phy reset is missing
[    0.392955] pinctrl-single 1804002c.pinmux: 544 pins, size 68
[    0.400072] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.407423] console [ttyS0] disabled
[    0.411300] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 2500000) is a 16550A
[    0.420495] console [ttyS0] enabled
[    0.427902] bootconsole [early0] disabled
[    0.442650] m25p80 spi0.0: mx25l12805d (16384 Kbytes)
[    0.447815] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.454313] Creating 4 MTD partitions on "spi0.0":
[    0.459198] 0x000000000000-0x000000040000 : "u-boot"
[    0.464922] 0x000000040000-0x000000050000 : "u-boot-env"
[    0.471030] 0x000000050000-0x000000ff0000 : "oem-firmware"
[    0.477242] 0x000000ff0000-0x000001000000 : "art"
[    0.486844] nand: device found, Manufacturer ID: 0x01, Chip ID: 0xf1
[    0.493335] nand: AMD/Spansion NAND 128MiB 3,3V 8-bit
[    0.498456] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.506173] Scanning device for bad blocks
[    0.514994] random: fast init done
[    0.608130] 3 fixed-partitions partitions found on MTD device ar934x-nand
[    0.615028] Creating 3 MTD partitions on "ar934x-nand":
[    0.620341] 0x000000000000-0x000000500000 : "kernel"
[    0.626127] 0x000000500000-0x000006000000 : "ubi"
[    0.631983] 0x000006000000-0x000008000000 : "oem-backup"
[    0.638761] libphy: Fixed MDIO Bus: probed
[    0.980920] libphy: ag71xx_mdio: probed
[    0.987578] switch0: Atheros AR8337 rev. 2 switch registered on mdio.0
[    1.638467] ag71xx 19000000.eth: connected to PHY at mdio.0:00 [uid=004dd036, driver=Atheros AR8216/AR8236/AR8316]
[    1.649611] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: rgmii
[    1.990786] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    2.000468] eth1: Atheros AG71xx at 0xba000000, irq 5, mode: sgmii
[    2.008804] NET: Registered protocol family 10
[    2.017954] Segment Routing with IPv6
[    2.021825] NET: Registered protocol family 17
[    2.026393] 8021q: 802.1Q VLAN Support v1.8
[    2.034432] UBI: auto-attach mtd5
[    2.037822] ubi0: attaching mtd5
[    2.877541] ubi0: scanning is finished
[    2.892778] ubi0: attached mtd5 (name "ubi", size 91 MiB)
[    2.898261] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    2.905272] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    2.912174] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    2.919252] ubi0: good PEBs: 728, bad PEBs: 0, corrupted PEBs: 0
[    2.925346] ubi0: user volume: 2, internal volumes: 1, max. volumes count: 128
[    2.932680] ubi0: max/mean erase counter: 2/0, WL threshold: 4096, image sequence number: 1583271448
[    2.941954] ubi0: available PEBs: 0, total reserved PEBs: 728, PEBs reserved for bad PEB handling: 20
[    2.951379] ubi0: background thread "ubi_bgt0d" started, PID 340
[    2.958537] block ubiblock0_0: created from ubi0:0(rootfs)
[    2.964156] ubiblock: device ubiblock0_0 (rootfs) set to be root filesystem
[    2.977481] VFS: Mounted root (squashfs filesystem) readonly on device 254:0.
[    2.989437] Freeing unused kernel memory: 1308K
[    2.994035] This architecture does not have kernel memory protection.
[    3.000596] Run /sbin/init as init process
[    3.500986] init: Console is alive
[    3.504632] init: - watchdog -
[    5.094204] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    5.229660] usbcore: registered new interface driver usbfs
[    5.235312] usbcore: registered new interface driver hub
[    5.240814] usbcore: registered new device driver usb
[    5.286388] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    5.302834] SCSI subsystem initialized
[    5.311378] ehci-platform: EHCI generic platform driver
[    5.316893] ehci-platform 1b000000.usb: EHCI Host Controller
[    5.322721] ehci-platform 1b000000.usb: new USB bus registered, assigned bus number 1
[    5.330753] ehci-platform 1b000000.usb: irq 14, io mem 0x1b000000
[    5.359183] ehci-platform 1b000000.usb: USB 2.0 started, EHCI 1.00
[    5.366176] hub 1-0:1.0: USB hub found
[    5.370405] hub 1-0:1.0: 1 port detected
[    5.374948] ehci-platform 1b400000.usb: EHCI Host Controller
[    5.380773] ehci-platform 1b400000.usb: new USB bus registered, assigned bus number 2
[    5.388795] ehci-platform 1b400000.usb: irq 15, io mem 0x1b400000
[    5.419183] ehci-platform 1b400000.usb: USB 2.0 started, EHCI 1.00
[    5.426258] hub 2-0:1.0: USB hub found
[    5.430445] hub 2-0:1.0: 1 port detected
[    5.438830] usbcore: registered new interface driver usb-storage
[    5.445843] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    5.456114] init: - preinit -
[    5.917520] usb 1-1: new high-speed USB device number 2 using ehci-platform
[    6.286414] random: procd: uninitialized urandom read (4 bytes read)
[    6.293343] usb-storage 1-1:1.0: USB Mass Storage device detected
[    6.308491] scsi host0: usb-storage 1-1:1.0
[    6.540956] random: jshn: uninitialized urandom read (4 bytes read)
[    6.655695] random: jshn: uninitialized urandom read (4 bytes read)
[    7.320886] scsi 0:0:0:0: Direct-Access     Generic  STORAGE DEVICE   0903 PQ: 0 ANSI: 6
[    7.688312] sd 0:0:0:0: [sda] 31116288 512-byte logical blocks: (15.9 GB/14.8 GiB)
[    7.697298] sd 0:0:0:0: [sda] Write Protect is off
[    7.702205] sd 0:0:0:0: [sda] Mode Sense: 21 00 00 00
[    7.703425] sd 0:0:0:0: [sda] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
[    7.722665]  sda: sda1
[    7.729682] sd 0:0:0:0: [sda] Attached SCSI removable disk
[    7.926911] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
[    7.932956] IPv6: ADDRCONF(NETDEV_UP): eth1.1: link is not ready
[    9.000336] eth1: link up (1000Mbps/Full duplex)
[    9.006126] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
[    9.017668] IPv6: ADDRCONF(NETDEV_CHANGE): eth1.1: link becomes ready
[   12.212659] mount_root: loading kmods from internal overlay
[   12.253688] kmodloader: loading kernel modules from //etc/modules-boot.d/*
[   12.261876] kmodloader: done loading kernel modules from //etc/modules-boot.d/*
[   12.384275] random: crng init done
[   12.387739] random: 3 urandom warning(s) missed due to ratelimiting
[   12.851921] sd 0:0:0:0: [sda] tag#0 UNKNOWN(0x2003) Result: hostbyte=0x00 driverbyte=0x08
[   12.860258] sd 0:0:0:0: [sda] tag#0 Sense Key : 0xb [current] 
[   12.866182] sd 0:0:0:0: [sda] tag#0 ASC=0x47 ASCQ=0x1 
[   12.871431] sd 0:0:0:0: [sda] tag#0 CDB: opcode=0x28 28 00 00 00 00 20 00 00 40 00
[   12.879120] print_req_error: I/O error, dev sda, sector 32
[   12.957747] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 509
[   13.062836] UBIFS (ubi0:1): recovery needed
[   13.307383] UBIFS (ubi0:1): recovery completed
[   13.312029] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
[   13.320001] UBIFS (ubi0:1): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[   13.330071] UBIFS (ubi0:1): FS size: 74280960 bytes (70 MiB, 585 LEBs), journal size 3682304 bytes (3 MiB, 29 LEBs)
[   13.340658] UBIFS (ubi0:1): reserved for root: 3508473 bytes (3426 KiB)
[   13.347367] UBIFS (ubi0:1): media format: w4/r0 (latest is w5/r0), UUID E7F101CC-C6F4-4AE7-9297-F32AA852420A, small LPT model
[   13.360804] block: attempting to load /tmp/ubifs_cfg/upper/etc/config/fstab
[   13.368089] block: unable to load configuration (fstab: Entry not found)
[   13.375023] block: attempting to load /tmp/ubifs_cfg/etc/config/fstab
[   13.381737] block: unable to load configuration (fstab: Entry not found)
[   13.388609] block: attempting to load /etc/config/fstab
[   13.395486] block: unable to load configuration (fstab: Entry not found)
[   13.402417] block: no usable configuration
[   13.406670] UBIFS (ubi0:1): un-mount UBI device 0
[   13.411501] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" stops
[   13.422341] UBIFS (ubi0:1): background thread "ubifs_bgt0_1" started, PID 512
[   13.637268] UBIFS (ubi0:1): UBIFS: mounted UBI device 0, volume 1, name "rootfs_data"
[   13.645254] UBIFS (ubi0:1): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[   13.655326] UBIFS (ubi0:1): FS size: 74280960 bytes (70 MiB, 585 LEBs), journal size 3682304 bytes (3 MiB, 29 LEBs)
[   13.665915] UBIFS (ubi0:1): reserved for root: 3508473 bytes (3426 KiB)
[   13.672643] UBIFS (ubi0:1): media format: w4/r0 (latest is w5/r0), UUID E7F101CC-C6F4-4AE7-9297-F32AA852420A, small LPT model
[   14.164660] block: attempting to load /tmp/ubifs_cfg/upper/etc/config/fstab
[   14.171922] block: unable to load configuration (fstab: Entry not found)
[   14.178843] block: attempting to load /tmp/ubifs_cfg/etc/config/fstab
[   14.185589] block: unable to load configuration (fstab: Entry not found)
[   14.192526] block: attempting to load /etc/config/fstab
[   14.197976] block: unable to load configuration (fstab: Entry not found)
[   14.205069] block: no usable configuration
[   14.210216] mount_root: overlay filesystem has not been fully initialized yet
[   14.217815] mount_root: switching to ubifs overlay
[   14.259581] overlayfs: upper fs does not support xattr, falling back to index=off and metacopy=off.
[   14.271535] urandom-seed: Seed file not found (/etc/urandom.seed)
[   14.376269] eth1: link down
[   14.395935] procd: - early -
[   14.398945] procd: - watchdog -
[   15.076492] procd: - watchdog -
[   15.079994] procd: - ubus -
[   15.147820] procd: - init -
[   15.947205] kmodloader: loading kernel modules from /etc/modules.d/*
[   16.111432] tun: Universal TUN/TAP device driver, 1.6
[   16.133126] gre: GRE over IPv4 demultiplexor driver
[   16.151035] ip_gre: GRE over IPv4 tunneling driver
[   16.227399] Loading modules backported from Linux version v5.4-rc8-0-gaf42d3466bdc
[   16.235122] Backport generated by backports.git v5.4-rc8-1-0-g368e8c51
[   16.415988] usbcore: registered new interface driver usblp
[   16.435255] wireguard: WireGuard 0.0.20200215 loaded. See www.wireguard.com for information.
[   16.443858] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[   16.493527] urngd: v1.0.2 started.
[   16.503340] xt_time: kernel timezone is -0000
[   17.002301] PPP generic driver version 2.4.2
[   17.020413] NET: Registered protocol family 24
[   17.175282] ath10k 5.1 driver, optimized for CT firmware, probing pci device: 0x3c.
[   17.199827] ath10k_pci 0000:00:00.0: enabling device (0000 -> 0002)
[   17.208384] ath10k_pci 0000:00:00.0: pci irq legacy oper_irq_mode 1 irq_mode 0 reset_mode 0
[   17.662878] firmware ath10k!fwcfg-pci-0000:00:00.0.txt: firmware_loading_store: map pages failed
[   18.016155] firmware ath10k!pre-cal-pci-0000:00:00.0.bin: firmware_loading_store: map pages failed
[   18.409416] firmware ath10k!QCA988X!hw2.0!ct-firmware-5.bin: firmware_loading_store: map pages failed
[   18.581040] firmware ath10k!QCA988X!hw2.0!ct-firmware-2.bin: firmware_loading_store: map pages failed
[   18.753003] firmware ath10k!QCA988X!hw2.0!firmware-6.bin: firmware_loading_store: map pages failed
[   18.923577] firmware ath10k!QCA988X!hw2.0!firmware-5.bin: firmware_loading_store: map pages failed
[   19.094915] firmware ath10k!QCA988X!hw2.0!firmware-4.bin: firmware_loading_store: map pages failed
[   19.265856] firmware ath10k!QCA988X!hw2.0!firmware-3.bin: firmware_loading_store: map pages failed
[   19.278439] ath10k_pci 0000:00:00.0: qca988x hw2.0 target 0x4100016c chip_id 0x043202ff sub 0000:0000
[   19.287843] ath10k_pci 0000:00:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   19.300281] ath10k_pci 0000:00:00.0: firmware ver 10.1-ct-8x-__fW-022-b8a7ba1b api 2 features wmi-10.x,has-wmi-mgmt-tx,mfp,txstatus-noack,wmi-10.x-CT,ratemask-CT,txrate-CT,get-temp-CT,tx-rc-CT,cust-stats-CT,retry-gt2-CT,txrate2-CT,beacon-cb-CT,wmi-block-ack-CT crc32 c8e2cb88
[   19.509342] firmware ath10k!QCA988X!hw2.0!board-2.bin: firmware_loading_store: map pages failed
[   19.518616] ath10k_pci 0000:00:00.0: board_file api 1 bmi_id N/A crc32 bebc7c08
[   20.450534] ath10k_pci 0000:00:00.0: unsupported HTC service id: 1536
[   20.457339] ath10k_pci 0000:00:00.0: 10.1 wmi init: vdevs: 16  peers: 127  tid: 256
[   20.474790] ath10k_pci 0000:00:00.0: wmi print 'P 128 V 8 T 410'
[   20.481132] ath10k_pci 0000:00:00.0: wmi print 'msdu-desc: 1424  sw-crypt: 0 ct-sta: 0'
[   20.489290] ath10k_pci 0000:00:00.0: wmi print 'alloc rem: 20904 iram: 25912'
[   20.543266] ath10k_pci 0000:00:00.0: htt-ver 2.1 wmi-op 2 htt-op 2 cal file max-sta 128 raw 0 hwcrypto 1
[   20.557544] ath10k_pci 0000:00:00.0: NOTE:  Firmware DBGLOG output disabled in debug_mask: 0x10000000
[   20.692298] ath: EEPROM regdomain: 0x0
[   20.692304] ath: EEPROM indicates default country code should be used
[   20.692308] ath: doing EEPROM country->regdmn map search
[   20.692320] ath: country maps to regdmn code: 0x3a
[   20.692325] ath: Country alpha2 being used: US
[   20.692328] ath: Regpair used: 0x3a
[   20.759438] batman_adv: B.A.T.M.A.N. advanced 2019.5-openwrt-0 (compatibility version 15) loaded
[   20.865286] ath: EEPROM regdomain: 0x0
[   20.865293] ath: EEPROM indicates default country code should be used
[   20.865296] ath: doing EEPROM country->regdmn map search
[   20.865309] ath: country maps to regdmn code: 0x3a
[   20.865314] ath: Country alpha2 being used: US
[   20.865317] ath: Regpair used: 0x3a
[   20.882880] ieee80211 phy1: Selected rate control algorithm 'minstrel_ht'
[   20.884294] ieee80211 phy1: Atheros AR9550 Rev:0 mem=0xb8100000, irq=13
[   20.894403] kmodloader: done loading kernel modules from /etc/modules.d/*
[   45.951607] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
[   45.958685] eth1: link up (1000Mbps/Full duplex)
[   45.979365] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
[   45.993054] br-lan: port 1(eth1.1) entered blocking state
[   45.998538] br-lan: port 1(eth1.1) entered disabled state
[   46.004363] device eth1.1 entered promiscuous mode
[   46.009360] device eth1 entered promiscuous mode
[   46.060511] br-lan: port 1(eth1.1) entered blocking state
[   46.065999] br-lan: port 1(eth1.1) entered forwarding state
[   46.071866] IPv6: ADDRCONF(NETDEV_UP): br-lan: link is not ready
[   46.154196] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
[   46.190253] IPv6: ADDRCONF(NETDEV_UP): eth0.2: link is not ready
[   46.999497] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   47.240504] eth0: link up (1000Mbps/Full duplex)
[   47.245266] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   47.279382] IPv6: ADDRCONF(NETDEV_CHANGE): eth0.2: link becomes ready
```

981213 commented:
Merged into my staging tree at: https://git.openwrt.org/?p=openwrt/staging/981213.git
Thanks!
